### PR TITLE
CORE-5939 crypto config schema

### PR DESCRIPTION
**Pull request by alexey-kadyrov-r3**
Defined the new crypto schema

### CODE REVIEW

File: data/config-schema/src/main/resources/net/corda/schema/configuration/cryptoLibrary/1.0/corda.cryptoLibrary.json [hunk 1]
```diff
@@ -5,6 +5,406 @@
   "description": "Configuration schema for the crypto library subsection.",
   "type": "object",
   "properties": {
-    "additionalProperties": true
-  }
+    "cryptoConnectionFactory": {
+      "description": "Settings for database connections factory's cache of EntityManagerFactory(s)",
+      "type": "object",
+      "default": {},
+      "properties": {
```
Comment (charlieR3) [inline data/config-schema/src/main/resources/net/corda/schema/configuration/cryptoLibrary/1.0/corda.cryptoLibrary.json:12]: Could nest this under `cache` to align with `signingService`

File: data/config-schema/src/main/resources/net/corda/schema/configuration/cryptoLibrary/1.0/corda.cryptoLibrary.json [hunk 2]
```diff
@@ -5,6 +5,406 @@
...
+          "type": "object",
+          "default": {},
+          "properties": {
+            "maxAttempts": {
+              "description": "Maximum attempts to process a message",
+              "type": "integer",
+              "default": 3
+            },
+            "waitBetweenMills": {
+              "description": "Timeouts between attempts in milliseconds, if the number of value less than intervals then the last item is repeated",
```
Comment (charlieR3) [inline data/config-schema/src/main/resources/net/corda/schema/configuration/cryptoLibrary/1.0/corda.cryptoLibrary.json]:
```suggestion
              "description": "Time between attempts in milliseconds, if the number of values is less than attempts then the last item is repeated",
```

File: data/config-schema/src/main/resources/net/corda/schema/configuration/cryptoLibrary/1.0/corda.cryptoLibrary.json [hunk 3]
```diff
@@ -5,6 +5,406 @@
...
```
Comment (charlieR3) [inline data/config-schema/src/main/resources/net/corda/schema/configuration/cryptoLibrary/1.0/corda.cryptoLibrary.json]:
```suggestion
              "description": "Time between attempts in milliseconds, if the number of values is less than attempts then the last item is repeated",
```

File: data/config-schema/src/main/resources/net/corda/schema/configuration/cryptoLibrary/1.0/corda.cryptoLibrary.json [hunk 4]
```diff
@@ -5,6 +5,406 @@
...
```
Comment (charlieR3) [inline data/config-schema/src/main/resources/net/corda/schema/configuration/cryptoLibrary/1.0/corda.cryptoLibrary.json]:
```suggestion
              "description": "Time between attempts in milliseconds, if the number of values is less than attempts then the last item is repeated",
```

File: data/config-schema/src/main/resources/net/corda/schema/configuration/cryptoLibrary/1.0/corda.cryptoLibrary.json [hunk 5]
```diff
@@ -5,6 +5,406 @@
...
+                "name",
+                "masterKeyPolicy",
+                "capacity",
+                "supportedSchemes",
+                "categories",
+                "cfg"
+              ]
+            }
+          },
+          "additionalProperties": false        }
```
Comment (charlieR3) [inline data/config-schema/src/main/resources/net/corda/schema/configuration/cryptoLibrary/1.0/corda.cryptoLibrary.json:405]: Formatting

**Review by charlieR3 — DISMISSED** [review]
Some comments added. Approved as I don't think they're major.

File: data/config-schema/src/main/resources/net/corda/schema/configuration/cryptoLibrary/1.0/corda.cryptoLibrary.json [hunk 6]
```diff
@@ -5,6 +5,406 @@
...
+        "expireAfterAccessMins": {
+          "description": "Expiration time in minutes for cached EntityManagerFactory",
+          "type": "integer",
+          "default": 5
+        },
+        "maximumSize": {
+          "description": "Maximum number of cached EntityManagerFactory",
+          "type": "integer",
+          "default": 3
+        }
```
Comment (driessamyn) [inline data/config-schema/src/main/resources/net/corda/schema/configuration/cryptoLibrary/1.0/corda.cryptoLibrary.json:22]: Slightly separate to this PR actually, but I don't think caching of db connections and EMFs should happen at the component level. Instead we should do this centrally as the same connections are used by multiple components.
So I'm not too keen on having this on the config schema.

File: data/config-schema/src/main/resources/net/corda/schema/configuration/cryptoLibrary/1.0/corda.cryptoLibrary.json (hunk 1 above)
Comment (alexey-kadyrov-r3) [reply data/config-schema/src/main/resources/net/corda/schema/configuration/cryptoLibrary/1.0/corda.cryptoLibrary.json:12]: The service which provides the connections is more global than the signing service only, it's used in other places

**Review by charlieR3 — APPROVED** [review]

File: data/config-schema/src/main/resources/net/corda/schema/configuration/cryptoLibrary/1.0/corda.cryptoLibrary.json (hunk 6 above)
Comment (driessamyn) [reply data/config-schema/src/main/resources/net/corda/schema/configuration/cryptoLibrary/1.0/corda.cryptoLibrary.json:22]: CC @mnesbit , WDYT ^ ?

My worry is if components each have their own cache of connections, it'll be very hard to tune this as vnode connections, for example, are used across different components, so all these caches will be fighting eachother when a central cache would make more sense.

Comment (alexey-kadyrov-r3) [reply data/config-schema/src/main/resources/net/corda/schema/configuration/cryptoLibrary/1.0/corda.cryptoLibrary.json:22]: I would prefer that to be central as well

**Pull request merged**